internal/db/sqlite/db_open_cgo.go

```diff
@@ -0,0 +1,17 @@
+//go:build cgo
+
+package sqlite
+
+import (
+	_ "github.com/mattn/go-sqlite3" // register sqlite3 database driver
+	"github.com/syncthing/syncthing/lib/build"
+)
+
+const (
+	dbDriver      = "sqlite3"
+	commonOptions = "_fk=true&_rt=true&_cslike=true&_cache_size=-65536&_sync=1"
```

Comment:
```
_cslike=true
```

https://www.sqlite.org/pragma.html#pragma_case_sensitive_like

> This pragma is deprecated and exists for backwards compatibility only. New applications should avoid using this pragma. Older applications should discontinue use of this pragma at the earliest opportunity. This pragma may be omitted from the build when SQLite is compiled using [SQLITE_OMIT_DEPRECATED](https://www.sqlite.org/compile.html#omit_deprecated).

Reply:
The `GLOB` operator is a possible alternative to using `LIKE`, as it is case-sensitive (both GLOB and LIKE also support wildcards, though their syntax is slightly different)

Reply:
Yes, I was just looking into that, and GLOB is better in that it almost does what we need, however it does not have any escaping that I can see. (So will mishandle prefixes that contain `*` for example, when `*` is part of the filename.)

Looks like it can be fixed with `>=` and `<` comparisons though.

Reply:
I have only glanced over the relevant code, but `instr()` (https://www.sqlite.org/lang_corefunc.html#instr) could be another possible substitute for LIKE, avoiding all the escape stuff.